test/test_config_mixin.py

```diff
@@ -68,3 +68,36 @@ def test_config_mixin() -> None:
     assert isinstance(model.data, Dataclass)
     assert model.data.x == 2
     assert model.data.y == 3
+
+    cfg = asdict(cfg)
+
+    model = Module.from_config(cfg)
+    assert isinstance(model, Module)
+    assert model.x == 0
+    # TODO: Support recursively restoring objects when cfg is dict.
+    # assert isinstance(model.data, Dataclass)
+    # assert model.data.x == 1
+    # assert model.data.y == 2
```

Comment:
Haven't taken a deeper look, but ideally we'd want to support recursively restoring objects.